instant-xml-macros/src/de.rs

```diff
@@ -624,10 +642,21 @@ fn named_field<'a>(
         } else if field_meta.direct {
             direct.extend(quote!(
                 Node::Text(text) => {
+                    seen_direct = true;
```

Comment:
Does this result in different handling between self-closing tags and empty text nodes? Is that desirable?

Reply:
The behavior is the same for both; I've added a couple of test cases to demonstrate that.
I think having them behave consistently with each other is reasonable.
I don't need to make a distinction in my downstream code.

I imagine that if you did care about the difference, you would want the direct value to be `Option<String>`:

```rust
#[derive(Debug, PartialEq, FromXml)]
struct OptionArtUri {
    #[xml(direct)]
    uri: Option<String>,
}
```

with the current state in this PR, we'll produce `Ok(OptionContainer { art: Some(OptionArtUri { uri: Some("") }) })` for that case.

I'm not sure if there's a reasonable way to make that come back as `uri: None` without making more significant changes to the traits and types, and/or having some understanding about whether the target type is `Option`.

Without this PR, this case produces a `MissingValue` error, so I don't think that things are worse off than they were.

Reply:
Agree that this behavior is desired.